File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.cc
#include <bits/stdc++.h>
using namespace std;
int cnt;
string s,first,second,middle;
map<char,int> mp;
int main()
{
	cin>>s;
	for(int i=0;i<s.size();i++)
	{
		mp[s[i]]++;
	}
	for(auto it : mp)
	{
		if(it.second%2==1)
		{
			cnt++;
			middle=it.first;
		}
		for(int i=0;i<it.second/2;i++)
		{
			first+=it.first;
		}
	}
	second = first;
	reverse(second.begin(),second.end());
	if(cnt>1)cout<<"I'm Sorry Hansoo";
	else cout<<first+middle+second;
}

File: 백준/Silver/1213. 팰린드롬 만들기/팰린드롬 만들기.py
from collections import defaultdict
line=input().strip()
mp=defaultdict(int)

cnt=0 # 홀수 카운팅
middle=''
first=''
second=''

for ch in line:
    mp[ch]+=1 # {'A' : 3, 'B' : 2}

for ch in sorted(mp.keys()):
    if mp[ch]%2==1:
        cnt+=1
        middle=ch
    for _ in range(mp[ch]//2):
        first+=ch

second=first[::-1]
if cnt>1:
    print("I'm Sorry Hansoo")
else:
    print(f"{first}{middle}{second}")